app/models/user.rb

```diff
@@ -9,4 +9,40 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :requested_friendships, foreign_key: 'requester_id', class_name: 'Friendship'
+  has_many :received_friendships, foreign_key: 'receiver_id', class_name: 'Friendship'
+
+  #  has_many :friendships
+  #  has_many :inverse_friendships, :class_name => "Friendship", :foreign_key => "friend_id"
+
+  def friends
+    requested_array = requested_friendships.map { |fr| fr.receiver if fr.status }
+    received_array = received_friendships.map { |fr| fr.requester if fr.status }
+    (requested_array + received_array).compact
+  end
+
+  def friendships
+    requested = requested_friendships
+    received = received_friendships
+    (requested + received)
+  end
+
+  def pending_friends
+    requested_friendships.map { |fr| fr.receiver unless fr.status }.compact
+  end
+
+  def friend_requests
+    received_friendships.map { |fr| fr.requester unless fr.status }.compact
+  end
+
+  def confirm_friend(user)
+    friendship = received_friendships.find { |fr| fr.requester == user }
+    friendship.status = true
+    friendship.save
+  end
```

Comment:
- [ ] This is the method you'll need to update, here instead of just updating the status to true you should also create a new Friendship in which the status will be true and the requester will be the receiver and vice-versa.